File: Problem 021/Solution 21.rb
# Solves http://projecteuler.net/problem=21

def d(num)
	sum = 0
	for n in (1..num-1)
		sum += n if(num % n == 0)
	end

	return sum
end

def checknum(num)
	n1 = d(num)
	n2 = d(n1)

	if num == n2 && n1 != n2
		return num
	else
		return 0
	end
end

sum = 0

for i in (1..10000)
	sum += checknum(i)
end
print "Answer: ", sum, "\n"

File: Problem 021/Solution 21.c
/* Solves: http://projecteuler.net/problem=21 */

#include <stdio.h>

int d(int num)
{
	int sum = 0;
	int i = 1;

	for(i = 1; i <= num-1; i++)
	{
		if (num % i == 0)
			sum += i;
	}

	return sum;
}

int checknum(int num)
{
	int n1 = d(num);
	int n2 = d(n1);

	if ((num == n2) && (n1 != n2))
	{
		return num;
	}
	else {
		return 0;
	}
}


int main()
{
	int i = 1;
	int sum = 0;
	for(i = 1; i <= 10000; i++)
	{
		sum += checknum(i);
	}

	printf("Answer: %d\r\n", sum);
}